db/view/view.cc

```diff
@@ -142,22 +142,63 @@ void view_info::set_base_info(db::view::base_info_ptr base_info) {
     _base_info = std::move(base_info);
 }
 
+// A constructor for a read and write enabled base info
+db::view::base_dependent_view_info::base_dependent_view_info(schema_ptr base_schema, std::vector<column_id>&& base_non_pk_columns_in_view_pk)
+        : base_schema{std::move(base_schema)}
+        , _base_non_pk_columns_in_view_pk{std::move(base_non_pk_columns_in_view_pk)}
+        , has_base_non_pk_columns_in_view_pk{!_base_non_pk_columns_in_view_pk.empty()}
+        , base_info_is_for_read_only{false} {
+
+}
+    // A constructor for a read only base info
+db::view::base_dependent_view_info::base_dependent_view_info(schema_ptr base_schema, bool has_base_non_pk_columns_in_view_pk)
+        : base_schema{std::move(base_schema)}
+        , has_base_non_pk_columns_in_view_pk{has_base_non_pk_columns_in_view_pk}
+        , base_info_is_for_read_only{true} {
+}
+
 db::view::base_info_ptr view_info::make_base_dependent_view_info(const schema& base) const {
     std::vector<column_id> base_non_pk_columns_in_view_pk;
+    bool has_base_non_pk_columns_in_view_pk = false;
+    bool base_info_is_for_read_only = false;
+    bool is_secondary_index = false;
```

Comment:
This is_secondary_index is not used anywhere?

Reply:
It is not used, I forgot to delete it.
I will.